File: desktop/macos/Sources/NexusDesktop/Bridge/DesktopRuntimeScript.swift
import Foundation

enum DesktopRuntimeScript {
  static func make(runtime: SidecarRuntimeConfig) throws -> String {
    let payload: [String: String] = [
      "api_base_url": runtime.apiBaseURL.absoluteString,
      "ws_url": runtime.webSocketURL.absoluteString,
      "auth_token": runtime.sessionToken,
      "app_mode": runtime.appMode,
      "app_version": runtime.appVersion,
      "build_number": runtime.buildNumber,
      "platform": runtime.platform,
    ]
    let data = try JSONSerialization.data(withJSONObject: payload, options: [])
    guard let json = String(data: data, encoding: .utf8) else {
      throw DesktopShellError.invalidRuntimeConfig
    }
    return "window.__NEXUS_DESKTOP_RUNTIME__ = \(json);"
  }
}

File: desktop/windows/Nexus.Desktop/Bridge/DesktopRuntimeScript.cs
using System.Text.Json;
using Nexus.Desktop.Sidecar;

namespace Nexus.Desktop.Bridge;

internal static class DesktopRuntimeScript
{
    public static string Make(SidecarRuntimeConfig runtime)
    {
        var payload = new Dictionary<string, string>
        {
            ["api_base_url"] = runtime.ApiBaseUrl,
            ["ws_url"] = runtime.WebSocketUrl,
            ["auth_token"] = runtime.SessionToken,
            ["app_mode"] = runtime.AppMode,
            ["app_version"] = runtime.AppVersion,
            ["build_number"] = runtime.BuildNumber,
            ["platform"] = runtime.Platform,
        };
        return $"window.__NEXUS_DESKTOP_RUNTIME__ = {JsonSerializer.Serialize(payload)};";
    }
}
